code/game/gamemodes/miniantags/pulsedemon/pulsedemon.dm

```diff
@@ -803,6 +804,32 @@
 /mob/living/simple_animal/demon/pulse_demon/mob_has_gravity()
 	return TRUE
 
+/obj/item/organ/internal/heart/demon/pulse
+	name = "perpetual pacemaker"
+	desc = "It still beats furiously, thousands of bright lights shine within it."
+	color = COLOR_YELLOW
+
+/obj/item/organ/internal/heart/demon/pulse/Initialize(mapload)
+	. = ..()
+	set_light(20, 2, "#bbbb00")
+
+/obj/item/organ/internal/heart/demon/pulse/attack_self(mob/living/user)
+	. = ..()
+	user.drop_item()
+	insert(user)
+
+/obj/item/organ/internal/heart/demon/pulse/insert(mob/living/carbon/M, special, dont_remove_slot)
+	. = ..()
+	M.AddComponent(/datum/component/cross_shock, 30, 500, 2 SECONDS)
+	ADD_TRAIT(M, TRAIT_SHOCKIMMUNE, UNIQUE_TRAIT_SOURCE(src))
+	M.set_light(3, 2, "#bbbb00")
+
+/obj/item/organ/internal/heart/demon/pulse/remove(mob/living/carbon/M, special)
+	..()
+	qdel(M.GetComponent(/datum/component/cross_shock))
```

Comment:
Is there an organ removal signal we can use to make the component delete itself here, instead of doing this somewhat nasty getcomponent call?

Reply:
Done